src/map.c

```diff
@@ -3,752 +3,632 @@
  * "LICENSE" for information on usage and redistribution of this file.
  */
 
+/* This map implementation has undergone extensive modifications, heavily
+ * relying on the rb.h header file from jemalloc.
+ * The original rb.h file served as the foundation and source of inspiration
+ * for adapting and tailoring it specifically for this map implementation.
+ * Therefore, credit and sincere thanks are extended to jemalloc for their
+ * invaluable work.
+ * Reference:
+ *   https://github.com/jemalloc/jemalloc/blob/dev/include/jemalloc/internal/rb.h
+ */
+
 #include <assert.h>
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
 
 #include "map.h"
 
-struct map_internal {
-    struct map_node *head;
+/* TODO: Avoid relying on key_size and data_size */
+struct map_head_t {
```

Comment:
The suffix `_t` is used for typedef. Refine the naming. I think `map_head` is confusing.

Reply:
How about `rb_entry`?